pkg/oc/cli/admin/node/listpods.go

```diff
@@ -32,90 +30,56 @@ func (l *ListPodsOptions) Run() error {
 		return err
 	}
 
-	var printer kprinters.ResourcePrinter
-	if l.Options.CmdPrinterOutput {
-		printer = l.Options.CmdPrinter
-	} else {
-		printer, err = l.Options.GetPrintersByResource(schema.GroupVersionResource{Resource: "pod"}, true)
-		if err != nil {
-			return err
-		}
-	}
-
-	// determine if printer kind is json or yaml and modify output
-	// to combine all pod lists into a single list
-	if l.Options.CmdPrinterOutput {
-		errs := l.handleRESTOutput(nodes, printer)
-		return kerrors.NewAggregate(errs)
+	unifiedPodList := &corev1.PodList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "List",
+			APIVersion: "v1",
+		},
 	}
+	successMessage := bytes.NewBuffer([]byte{})
 
 	errList := []error{}
 	for _, node := range nodes {
-		err := l.runListPods(node, printer)
+		podList, err := l.podsForNode(node)
 		if err != nil {
-			// Don't bail out if one node fails
 			errList = append(errList, err)
+			continue
 		}
-	}
-	return kerrors.NewAggregate(errList)
-}
 
-func (l *ListPodsOptions) runListPods(node *kapi.Node, printer kprinters.ResourcePrinter) error {
-	labelSelector, err := labels.Parse(l.Options.PodSelector)
-	if err != nil {
-		return err
+		fmt.Fprintf(successMessage, "Listing matched pods on node %q:\n", node.ObjectMeta.Name)
+		for _, pod := range podList.Items {
+			fmt.Fprintf(successMessage, " - %s\n", pod.Name)
```

Comment:
@soltysh this used the "humanreadable" printer before. Now it just uses the "success" message of the new printer stack. We can either leave it as it currently is, and print only pod names for each node, or try to mimic the old "humanreadable" output and print a few columns per pod (still using success output), similar to how it's being done in the policy commands PR: https://github.com/openshift/origin/pull/20810/files#diff-d16c3a1720bc945e401df0ff7c782325R268

Reply:
Do the policy-like output.

Reply:
Thanks, done